operator/app/src/test/java/org/keycloak/operator/utils/CRAssert.java

```diff
@@ -32,13 +33,17 @@ public static void assertKeycloakStatusCondition(Keycloak kc, String condition,
         assertKeycloakStatusCondition(kc, condition, status, null);
     }
     public static void assertKeycloakStatusCondition(Keycloak kc, String condition, boolean status, String containedMessage) {
-        Log.infof("Asserting CR: %s, condition: %s, status: %s, message: %s", kc.getMetadata().getName(), condition, status, containedMessage);
-        assertThat(kc.getStatus().getConditions().stream()
-                .anyMatch(c ->
-                        c.getType().equals(condition) &&
-                        c.getStatus() == status &&
-                        (containedMessage == null || c.getMessage().contains(containedMessage)))
-                    ).isTrue();
+        try {
+            assertThat(kc.getStatus().getConditions().stream()
+                    .anyMatch(c ->
+                            c.getType().equals(condition) &&
+                                    c.getStatus() == status &&
+                                    (containedMessage == null || c.getMessage().contains(containedMessage)))
+            ).isTrue();
+        } catch (Exception e) {
+            Log.infof("Asserting CR: %s with status:\n%s", kc.getMetadata().getName(), Serialization.asYaml(kc.getStatus()));
```

Comment:
I know this is a matter of preference, but I liked that the log was printed during each poll in a waiting loop. Particularly when running on my machine it made the test look alive and it was much quickly visible what is going on, what we're waiting for (especially in the amount of other log messages). Maybe we could decrease the level of that log message instead? That would make it possible to hide it by default but visible when needed (local development).

Reply:
Are you proposing to keep the log on the top and assign it at the `trace` level for example?
Do you agree to keep this more detailed log in case of failure?

Reply:
Yes (except for `trace` I'd personally make it `debug`) and yes. 🙂

Reply:
done